src/core/components/IsolatedInput/index.tsx

```diff
@@ -38,6 +38,16 @@ export default class IsolatedInput extends PureComponent<IProps, IState> {
         return this.props as PropsWithDefaults;
     }
 
+    componentWillReceiveProps(nextProps: IProps) {
+        const { value: nextValue } = nextProps;
+
+        if (this.state.value !== nextValue) {
+            this.setState({
+                value: nextValue
+            });
+        }
+    }
+
```

Comment:
The input needs to update its internal state when the value/prop updates. That prevented the UI from updating to match the filter.